Automattic/wp-calypso
# State: Migrate lib/site-roles to Redux state

This PR migrates from the legacy [`lib/site-roles` Flux store](https://github.com/Automattic/wp-calypso/tree/master/client/lib/site-roles) to Redux - fixes #7921. The PR takes care of the following:

* Creates Redux actions, reducer, schema, selectors and README for site roles.
* Adds tests for the above.
* Introduces a `QuerySiteRoles` query component.
* Updates the `RoleSelect` component to use Redux and converts it to a stateless functional component.
* Removes the legacy `lib/site-roles` Flux store.

To test:

1. Checkout this branch.
2. Use one of your **public** WordPress.com sites for the next steps.
3. Go to **People**, click on a user and verify the roles load properly in the **Role** field.
4. Click on the button to add a new user and verify the roles load properly in the **Role** field.
5. Verify step 3 and 4 work properly with a clean Redux state.
6. Verify the last role in the role select field is called `Follower`.
7. Use one of your **private** WordPress.com sites, go through steps 3 to 6, but this time verify the last role is called "Viewer".
8. Use one of your Jetpack sites and go through step 3 to 5, verify all works as expected.
9. Verify all tests pass: `npm run test-client client/state/site-roles/`
